src/sentry/tasks/store.py

```diff
@@ -177,24 +183,66 @@ def delete_raw_event(project_id, event_id):
         event_id=event_id
     ).delete()
 
-
-def create_failed_event(cache_key, project, issues, event_id):
+    # Clear the sent notification if we reprocessed everything
+    # successfully and reprocessing is enabled
+    reprocessing_active = ProjectOption.objects.get_value(
+        project_id, 'sentry:reprocessing_active', REPROCESSING_DEFAULT)
+    if reprocessing_active:
+        sent_notification = ProjectOption.objects.get_value(
+            project_id, 'sentry:sent_failed_event_hint', False)
+        if sent_notification:
+            if ReprocessingReport.objects.filter(
+                project_id=project_id,
+                event_id=event_id
+            ).values('id').first() is None:
```

Comment:
ProTip:

`.exists()` translates to a `SELECT 1 ...` and is what you should use here which returns a boolean.